test/e2e/upgradetest/framework/framework.go

```diff
@@ -142,7 +142,10 @@ func (f *Framework) DeleteOperator(name string) error {
 		LabelSelector: labels.SelectorFromSet(operatorLabelSelector(name)).String(),
 	}
 	_, err = e2eutil.WaitPodsDeletedCompletely(f.KubeCli, f.KubeNS, 30*time.Second, lo)
-	return err
+	if err != nil {
+		return err
+	}
+	return f.KubeCli.CoreV1().Endpoints(f.KubeNS).Delete("etcd-operator", metav1.NewDeleteOptions(0))
```

Comment:
add a comment here? leader election might change to not use endpoint in the future. this deletion might fails with resource not found.